subsys/bluetooth/host/conn.c

```diff
@@ -1222,10 +1223,14 @@ void bt_conn_set_state(struct bt_conn *conn, bt_conn_state_t state)
 			 * the application through bt_conn_disconnect or by
 			 * timeout set by bt_conn_le_create_param.timeout.
 			 */
-			if (conn->err) {
-				notify_connected(conn);
+			if (IS_ENABLED(CONFIG_BT_CENTRAL)) {
+				bt_le_scan_update_and_reconfigure(
+					SCAN_ENABLED_REASON_SCAN_BEFORE_INITIATE,
+					BT_HCI_LE_SCAN_DISABLE);
```

Comment:
This is a change is behavior, is it not?

It looks like a welcome change, but why are we giving a scan enable reason when we disable?